src/h5web/dataset-visualizer/VisProvider.tsx

```diff
@@ -5,36 +5,59 @@ import React, {
   useContext,
   useMemo,
 } from 'react';
-import { DimensionMapping } from './models';
+import { transpose } from 'mathjs';
+import { isNumber } from 'lodash-es';
+import { DimensionMapping, Vis } from './models';
 import { HDF5Dataset, HDF5Value } from '../providers/models';
 import { useValue } from '../providers/hooks';
 import { isSimpleShape } from '../providers/utils';
 
 export interface VisProps {
-  mapping: DimensionMapping;
-  rawValues: HDF5Value;
+  visValues: HDF5Value;
   rawDims: number[];
+  slicingIndices?: number[];
```

Comment:
`slicingIndices` is now optional as we could imagine having only one dimension on `x`.